src/gsy_e/models/strategy/state/heatpump_tank_states/all_tanks_state.py

```diff
@@ -36,11 +36,11 @@ def __init__(self, tank_parameters: List[TankParameters]):
 
     def increase_tanks_temp_from_heat_energy(self, heat_energy_kJ: float, time_slot: DateTime):
         """Increase the temperature of the water tanks with the provided heat energy."""
-        scaling_factors = self._get_scaling_factors_for_charging_energy(
-            self._last_time_slot(time_slot)
-        )
+        scaling_factors = self._get_scaling_factors_for_charging(self._last_time_slot(time_slot))
         for num, tank in enumerate(self._tanks_states):
             heat_energy_per_tank_kWh = convert_kJ_to_kWh(heat_energy_kJ * scaling_factors[num])
+            if heat_energy_per_tank_kWh == 0:
```

Comment:
Same comment as above, tolerance comparison should be added. Same in the `decrease_tanks_temp_from_heat_energy` method